hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/TestDirectStoreSplitsMerges.java

```diff
@@ -202,6 +207,22 @@ public void testCommitMergedRegion() throws Exception {
     mergeRegionFs.commitMergedRegion();
   }
 
+  private void waitForSplitProcComplete(int attempts, int waitTime) throws Exception {
+    List<Procedure<?>> procedures = TEST_UTIL.getHBaseCluster().getMaster().getProcedures();
+    if(procedures.size()>0) {
+      Procedure splitProc = procedures.stream().
+        filter(p -> p instanceof SplitTableRegionProcedure).findFirst().get();
```

Comment:
Had to modify this logic a bit. The branch-2 version is a bit wrong, leading to NoSuchElement when the first proc is not a split one. I will cherry-pick this once it gets merged.